tuna/miopen/import_configs.py

```diff
@@ -156,87 +99,83 @@ def insert_config(driver, counts, dbt, args):
+              config=new_cf.id)
           session.add(new_cf_tag)
           session.commit()
           counts['cnt_tagged_configs'].add(new_cf.id)
       except IntegrityError as err:
-        LOGGER.warning("Err occurred: %s", err)
+        logger.warning("Err occurred: %s", err)
         session.rollback()
-    if args.mark_recurrent or args.tag:
-      _ = tag_config_v2(driver, counts, dbt, args, new_cf)
+    if args.import_configs.mark_recurrent or args.import_configs.tag:
+      _ = tag_config_v2(driver, counts, dbt, args, logger, new_cf)
 
   return new_cf.id
 
 
-def process_config_line_v2(driver, args, counts, dbt):
+def process_config_line_v2(driver, args, counts, dbt, logger):
   """Assumes config passed already exists and will skip the insert step
         if tag_only present. Otherwise it will first try and insert and
         then tag."""
-  if args.tag_only:
-    _ = tag_config_v2(driver, counts, dbt, args, new_cf=None)
+  if args.import_configs.tag_only:
+    _ = tag_config_v2(driver, counts, dbt, args, logger, new_cf=None)
     return False
 
-  _ = insert_config(driver, counts, dbt, args)
+  _ = insert_config(driver, counts, dbt, args, logger)
   return True
 
 
-def parse_line(args, line, counts, dbt):
+def parse_line(args, line, counts, dbt, logger):
   """parse a driver line or fdb line from an input file and insert the config"""
   if args.config_type == ConfigType.batch_norm:
-    driver = DriverBatchNorm(line, args.command)
+    driver = DriverBatchNorm(line, args.import_configs.command)
   else:
-    driver = DriverConvolution(line, args.command)
+    driver = DriverConvolution(line, args.import_configs.command)
 
-  if not args.batch_list:
-    process_config_line_v2(driver, args, counts, dbt)
+  if not args.import_configs.batch_list:
+    process_config_line_v2(driver, args, counts, dbt, logger)
   else:
-    for bsz in args.batch_list:
-      LOGGER.info('Batchsize: %s', bsz)
+    for bsz in args.import_configs.batch_list:
+      logger.info('Batchsize: %s', bsz)
       driver.batchsize = bsz
-      process_config_line_v2(driver, args, counts, dbt)
+      process_config_line_v2(driver, args, counts, dbt, logger)
 
   return True
 
 
-def import_cfgs(args, dbt):
+def import_cfgs(args, dbt, logger):
   """import configs to mysql from file with driver invocations"""
   connect_db()
 
   counts = {}
   counts['cnt_configs'] = 0
   counts['cnt_tagged_configs'] = set()
-  with open(os.path.expanduser(args.file_name), "r") as infile:  # pylint: disable=unspecified-encoding
+  with open(os.path.expanduser(args.import_configs.file_name), "r") as infile:  # pylint: disable=unspecified-encoding
     for line in infile:
       try:
-        parse_line(args, line, counts, dbt)
+        parse_line(args, line, counts, dbt, logger)
       except ValueError as err:
-        LOGGER.warning(err)
+        logger.warning(err)
         continue
 
   return counts
 
 
-def main():
```

Comment:
If we import the parser from parse_miopen_args.py, I think we can keep main and call run_import_configs after calling parse_args().

Reply:
we would need the parser from miopen_lib that contains the entire scope of the arguments not just the subcommand arguments.